service/src/main/java/com/codeforcommunity/processor/ProcessorImpl.java

```diff
@@ -46,14 +46,16 @@ public ProcessorImpl(DSLContext db) {
   }
 
   @Override
-  public boolean createNews(String title, String description, String author, LocalDateTime date, String content) {
+  public Optional<NewsReturn> createNews(String title, String description, String imageUrl, String author, LocalDateTime date, String content) {
     try {
-      db.insertInto(Tables.NEWS, Tables.NEWS.TITLE, Tables.NEWS.DESCRIPTION, Tables.NEWS.AUTHOR, Tables.NEWS.DATE,
-          Tables.NEWS.CONTENT).values(title, description, author, Timestamp.valueOf(date), content).execute();
+      Result result = db.insertInto(Tables.NEWS, Tables.NEWS.TITLE, Tables.NEWS.DESCRIPTION, Tables.NEWS.IMAGE_URL, Tables.NEWS.AUTHOR, Tables.NEWS.DATE,
+              Tables.NEWS.CONTENT).values(title, description, imageUrl, author, Timestamp.valueOf(date), content)
+              .returning(Tables.NEWS.ID)
+              .fetch();
+      return getNews((int) result.getValue(0,0));
     } catch (Exception e) {
```

Comment:
Nice use of optionals here, a step for further improvement would be to remove this catch Exception block. As a general rule you only want a catch statement to be as specific as possible and ideally you don't want to use them for normal execution (i.e. you shouldn't expect to catch an exception). If the point of this block is to catch database clashes for example you can just do a separate jOOQ call to see if there are any similar News objects and return the empty optional from there.